rollingpin/commands.py

```diff
@@ -0,0 +1,82 @@
+from abc import ABCMeta, abstractproperty
+
+
+class Command(object):
+    __metaclass__ = ABCMeta
+
+    CONTINUE = 1
+    SKIP_REMAINING = 2
+
+    def __init__(self, args=None):
+        if args is None:
+            self._args = []
+        else:
+            self._args = args
+
+    @abstractproperty
+    def name(self):
+        raise NotImplementedError
+
+    @property
+    def args(self):
+        return self._args
+
+    def add_argument(self, arg):
+        self._args.append(arg)
+
+    def cmdline(self):
+        return [self.name()] + self._args
+
+    def check_result(self, result):
+        return Command.CONTINUE
+
+    def __str__(self):
+        return "Command(name={}, args={})".format(self.name(), self._args)
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class SynchronizeCommand(Command):
+    def name(self):
+        return "synchronize"
+
+
+class DeployCommand(Command):
+    def name(self):
+        return "deploy"
+
+    def check_result(self, result):
+        # For backwards compatibility
```

Comment:
prior to this change, the deploy scripts will return a falsy dict -- in that case we want to fall back to the default behavior.